src/sparseml/transformers/finetune/text_generation.py

```diff
@@ -0,0 +1,343 @@
+    for training and eval
+    :param training_args: Arguments pertaining to training loop configuration
+    """
+
+    # Setup logging
+    log_level = training_args.get_process_log_level()
+    _LOGGER.setLevel(log_level)
+    datasets.utils.logging.set_verbosity(log_level)
+    transformers.utils.logging.set_verbosity(log_level)
+    transformers.utils.logging.enable_default_handler()
+    transformers.utils.logging.enable_explicit_format()
+
+    # Summary on each process
+    _LOGGER.warning(
+        f"Process rank: {training_args.local_rank}, device: {training_args.device}, "
+        f"n_gpu: {training_args.n_gpu}, "
+        f"distributed training: {bool(training_args.local_rank != -1)}, "
+        f"16-bits training: {training_args.fp16}"
+    )
+    _LOGGER.info(f"Training/evaluation parameters {training_args}")
+
+    # Detecting last checkpoint.
+    last_checkpoint = None
+    if (
+        os.path.isdir(training_args.output_dir)
+        and training_args.do_train
+        and not training_args.overwrite_output_dir
+    ):
+        last_checkpoint = get_last_checkpoint(training_args.output_dir)
+        if last_checkpoint is None and (len(os.listdir(training_args.output_dir)) > 0):
+            raise ValueError(
+                f"Output directory ({training_args.output_dir}) already "
+                "exists and is not empty. Use --overwrite_output_dir to overcome."
+            )
+        elif (
+            last_checkpoint is not None and training_args.resume_from_checkpoint is None
+        ):
+            _LOGGER.info(
+                f"Checkpoint detected, resuming training at {last_checkpoint}. To "
+                "avoid this behavior, change  the `--output_dir` or add "
+                "`--overwrite_output_dir` to train from scratch."
+            )
+
+    # Set seed before initializing model.
+    set_seed(training_args.seed)
+
+    # Load pretrained model
+    # The .from_pretrained methods guarantee that only one local process can
+    # concurrently download model & vocab.
+    config = AutoConfig.from_pretrained(
+        model_args.config_name
+        if model_args.config_name
+        else model_args.model_name_or_path,
+        cache_dir=model_args.cache_dir,
+        revision=model_args.model_revision,
+        use_auth_token=True if model_args.use_auth_token else None,
+    )
+
+    model_kwargs = {
+        "config": config,
+        "cache_dir": model_args.cache_dir,
+        "revision": model_args.model_revision,
+        "use_auth_token": True if model_args.use_auth_token else None,
+    }
+    # this calls from_pretrained under the hood so should be FSDP safe
+    model = SparseAutoModel.text_generation_from_pretrained(
+        model_name_or_path=model_args.model_name_or_path,
+        model_type="model",
+        **model_kwargs,
+    )
+
+    # initialize structure of input model from recipe if needed
+    model_path = model_args.model_name_or_path
```

Comment:
what is the relationship between this initialization and that in the mixin

Reply:
This is the same var passed to the mixin, it gets logged a couple times so I reassigned it for brevity. I'll update the mixin init to all use this var to make it more clear